metis/apfMETIS.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2025 Scientific Computation Research Center
+ *
+ * This work is open source software, licensed under the terms of the
+ * BSD license as described in the LICENSE file in the top-level directory.
+ */
+#ifndef APF_METIS_H
+#define APF_METIS_H
+
+#define APF_METIS_MAXRANKS 256
+
+/**
+ * \file apfMETIS.h
+ * \brief METIS partitioning for apf::Mesh objects.
+ */
+namespace apf {
+
+class Mesh;
+class Splitter;
+class Balancer;
+
+/**
+ * \addtogroup Partitioning
+ * \{
+ */
+
+/**
+ * \brief Make an apf::Splitter that calls METIS for the underlying algorithm
+ *
```

Comment:
```suggestion
 *\remark see makeMETISsplitter for additional details
```

Reply:
No localizing is done for the splitter since the whole mesh is already available.

Reply:
Well in c318dc92eebb51f69cc373e023256a7de2f0d7cd I respecified that msplit goes from 1-to-N ranks. Before that I that it was supposed to be like zsplit. I don't remember exactly why but I can probably change it back if preferred.

Reply:
I believe the migration plan would have the wrong numbers. I think you would need to PCU split modulo factor, then remapPartition with Unmodulo to get the desired n->factor*n result.